google-cloud-examples/src/main/java/com/google/cloud/examples/spanner/snippets/DatabaseAdminClientSnippets.java

```diff
@@ -53,7 +54,7 @@ public DatabaseAdminClientSnippets(DatabaseAdminClient dbAdminClient) {
   // [VARIABLE my_database_id]
   public Database createDatabase(String instanceId, String databaseId) {
     // [START createDatabase]
-    Operation<Database, CreateDatabaseMetadata> op = dbAdminClient
+    OperationFuture<Database, CreateDatabaseMetadata> op = dbAdminClient
```

Comment:
What happens when users cancel this future? Also if I just want to fire and forget and not poll for operation status, how would I do that?

Reply:
Isn't fire and forget a bit of an anti-pattern? You won't get notified if the operation fails. I think you can do
```java
ApiFutures.addCallback(client.createDatabase(...), new ApiCallback<Database>() {
  @Override public void onSuccess(Database db) {/*no op*/}
  @Override public void onFailure(Throwable t) {logger.log(t);}
});
```
to at least get notified.

@garrettjonesgoogle can confirm. AFAIK, cancelling the future stops the background polling job, but does not cancel the job on the server.

Reply:
Actually @vam-google might be a better person to confirm

Reply:
Yes, cancelling polling/retrying future does not cancel corresponding server task. This is by design, since there can be potentially 2+ polling jobs for same server task (in same or different processes/hosts), so cancelling one should not cancel the whole thing (i..e server-side and another future). Also a future on a client side can be cancelled or "broken" in any other way by programming mistake in the client-side business logic. Client side problems/bugs should not affect server-side execution. If somebody wants to cancel server side operation, a specific call must be made (which itself is retriable and uses same logic as retries/polls).